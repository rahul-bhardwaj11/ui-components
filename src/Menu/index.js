import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AntMenu from 'antd/lib/menu';
import 'antd/lib/menu/style/index.css';
import StringToHTML from '../StringToHTML';
import styled from 'styled-components';
import mixins from '../styles/mixins';
import theme from '../styles/theme';

const MtMenu = styled.div`
  .ant-menu {
    font-family: inherit;
  }
  .ant-anchor-wrapper {
    overflow: hidden;
  }
  .ant-select-dropdown-menu-item {
    background-color: ${theme.colors.INDIGO};
    color: ${theme.colors.WHITE};
  }
  .ant-select-dropdown {
    background-color: ${theme.colors.INDIGO};
    color: ${theme.colors.WHITE};
  }
  .ant-select-dropdown-menu {
    background-color: ${theme.colors.WHITE};
    color: ${theme.colors.GREY};
    padding: 0px 5px;
    margin: 8px;
    border-radius: 4px;
    height: 32px;
    line-height: 32px;
  }
  .ant-dropdown-menu {
    padding: 4px 0px;
  }
  .ant-dropdown-menu-item:first-child,
  .ant-dropdown-menu-item:last-child {
    border-radius: 4px;
  }
  .ant-dropdown-menu-submenu-title {
    -webkit-transition: all 0.2s;
    transition: all 0.2s;
    &:hover {
      background-color: ${theme.colors.INDIGO};
      color: ${theme.colors.WHITE};
    }
  }
  .ant-dropdown-menu-item {
    background-color: ${theme.colors.WHITE};
    color: ${theme.colors.OUTER_SPACE};
    padding: 0px 15px;
    margin: 4px 8px;
    border-radius: 4px;
    height: 32px;
    line-height: 32px;
    -webkit-transition: all 0.2s;
    transition: all 0.2s;
    &:hover {
      background-color: ${theme.colors.INDIGO};
      color: ${theme.colors.WHITE};
    }
  }
  .ant-menu-horizontal {
    line-height: normal;
    & > .ant-menu-item {
      &:hover {
        color: ${theme.colors.CONGRESS_BLUE};
        border-bottom: 3px solid transparent;
      }
    }
    & > .ant-menu-submenu {
      &:hover {
        color: ${theme.colors.CONGRESS_BLUE};
      }
    }
    & > .ant-menu-item-active,
    & > .ant-menu-submenu-active,
    & > .ant-menu-item-selected,
    & > .ant-menu-submenu-selected {
      color: ${theme.colors.CONGRESS_BLUE};
      border-bottom: 3px solid ${theme.colors.INDIGO};
    }
  }
  .ant-menu-vertical {
    & > .ant-menu-item {
      color: ${theme.colors.OUTER_SPACE};
      padding: 0px;
      margin: 0px;
      text-align: left;
      &:hover {
        color: ${theme.colors.CONGRESS_BLUE};
      }
    }
  }
  .ant-menu-item {
    color: ${theme.colors.OUTER_SPACE};
    padding: 0px 8px;

    &:hover {
      color: ${theme.colors.WHITE};
    }
    b {
      font-weight: normal;
    }
  }
  .ant-dropdown-trigger {
    font-size: 13px;
  }

  & {
    .ant-menu-inline > .ant-menu-submenu > .ant-menu-submenu-title {
      text-align: left;
      font-size: 11px;
      text-transform: uppercase;
      height: auto;
      line-height: 16px;
      margin: 0px 0px 8px 0px;
      color: ${theme.colors.OUTER_SPACE};

      &:hover {
        color: ${theme.colors.OUTER_SPACE};
      }

      .ant-menu-submenu-arrow {
        display: none;
      }
    }
  }
  & {
    .ant-menu-sub.ant-menu-inline > .ant-menu-item {
      float: left;
      text-align: left;
      margin: 0px;
      font-size: 12px;
    }
  }

  .ant-anchor-link {
    padding: 6px 0 6px 0px;
    line-height: 1;
    display: list-item;
    margin-left: 17px;
    list-style-type: disc;
    color: ${theme.colors.INDIGO};

    a {
      color: ${theme.colors.INDIGO};
      font-size: 12px;
      ${mixins.truncate('250px')};
    }
  }
  .ant-menu-inline,
  .ant-menu-light {
    border-right: 0px;
  }
  .cautious {
    color: ${theme.colors.BITTERSWEET};
    &:active {
      background-color: ${theme.colors.BITTERSWEET};
      color: ${theme.colors.WHITE};
    }
    &:hover {
      background-color: ${theme.colors.BITTERSWEET};
      color: ${theme.colors.WHITE};
    }
  }

  .ant-dropdown-menu-item-selected {
    background-color: ${theme.colors.TROPICAL_BLUE};
  }
  .ant-menu-vertical .ant-menu-item:not(:last-child),
  .ant-menu-vertical .ant-menu-item {
    margin-bottom: 0px;
  }
`;
class Menu extends Component {
  static propTypes = {
    options: PropTypes.array,
    onClick: PropTypes.func,
    mode: PropTypes.string,
    children: PropTypes.node,
    prefixCls: PropTypes.string,
    itemStyle: PropTypes.object
  };

  static defaultProps = {
    style: { paddingLeft: '0px' },
    onClick: () => {}
  };

  onClick = ({ key }) => {
    this.props.onClick(key);
  };

  render() {
    let { options, children, itemStyle } = this.props;
    return (
      <MtMenu>
        <AntMenu {...this.props} onClick={this.onClick}>
          {!options && children}
          {options &&
            options.map((option, index) => {
              return (
                <AntMenu.Item
                  key={option.key || index}
                  style={itemStyle}
                  className={option.cautious ? 'cautious' : ''}
                >
                  {typeof option.content === 'string' ? (
                    <StringToHTML content={option.content} />
                  ) : (
                    option.content
                  )}
                </AntMenu.Item>
              );
            })}
        </AntMenu>
      </MtMenu>
    );
  }
}

Menu.ItemGroup = AntMenu.ItemGroup;
Menu.SubMenu = AntMenu.SubMenu;
Menu.Item = AntMenu.Item;
export default Menu;
