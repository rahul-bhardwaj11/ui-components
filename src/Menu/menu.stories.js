import React from 'react';
import { storiesOf } from '@storybook/react';
import Menu from './index';
import { withInfo } from '@storybook/addon-info';
import { withKnobs, object, text } from '@storybook/addon-knobs';

const stories = storiesOf('Menu', module);
stories.addDecorator(withKnobs);

const options = [
  { key: '1', content: '<b>First Item</b>' },
  { key: '2', content: 'Second Item' },
  { key: '3', content: 'Third Item' }
];

stories
  .add(
    'Default Menu',
    withInfo('Basic usage of the Menu')(() => (
      <Menu
        options={object('options', options)}
        mode={text('mode', 'horizontal')}
      />
    ))
  )
  .add(
    'Vertical Menu',
    withInfo('Vertical Menu')(() => (
      <Menu
        options={object('options', options)}
        mode={text('mode', 'vertical')}
      />
    ))
  );
