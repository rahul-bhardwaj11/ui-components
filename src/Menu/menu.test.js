import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Menu from './index';

describe('Menu', () => {
  const options = [
    { key: '1', content: '<b>First Item</b>' },
    { key: '2', content: 'Second Item' },
    { key: '3', content: 'Third Item' }
  ];
  it('matches default snapshot', () => {
    const tree = renderer
      .create(<Menu options={options} mode="horizontal" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('matches Vertical Menu snapshot', () => {
    const tree = renderer
      .create(<Menu options={options} mode="vertical" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should handle click', () => {
    const clickSpy = jest.fn();
    const menu = shallow(<Menu options={options} onClick={clickSpy} />);
    menu.find('Menu').simulate('click', { key: '1' });
    expect(clickSpy).toHaveBeenCalled();
  });
});
