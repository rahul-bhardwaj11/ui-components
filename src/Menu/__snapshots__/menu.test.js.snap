// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu matches Vertical Menu snapshot 1`] = `
<div
  className="sc-bdVaJa edRcnR"
>
  <ul
    className="ant-menu ant-menu-light ant-menu-root ant-menu-vertical"
    options={
      Array [
        Object {
          "content": "<b>First Item</b>",
          "key": "1",
        },
        Object {
          "content": "Second Item",
          "key": "2",
        },
        Object {
          "content": "Third Item",
          "key": "3",
        },
      ]
    }
    role="menu"
    style={Object {}}
  >
    <li
      aria-disabled={undefined}
      className="ant-menu-item"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      role="menuitem"
      style={Object {}}
      title={undefined}
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<b>First Item</b>",
          }
        }
      />
    </li>
    <li
      aria-disabled={undefined}
      className="ant-menu-item"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      role="menuitem"
      style={Object {}}
      title={undefined}
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Second Item",
          }
        }
      />
    </li>
    <li
      aria-disabled={undefined}
      className="ant-menu-item"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      role="menuitem"
      style={Object {}}
      title={undefined}
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Third Item",
          }
        }
      />
    </li>
  </ul>
</div>
`;

exports[`Menu matches default snapshot 1`] = `
<div
  className="sc-bdVaJa edRcnR"
>
  <ul
    className="ant-menu ant-menu-light ant-menu-root ant-menu-horizontal"
    options={
      Array [
        Object {
          "content": "<b>First Item</b>",
          "key": "1",
        },
        Object {
          "content": "Second Item",
          "key": "2",
        },
        Object {
          "content": "Third Item",
          "key": "3",
        },
      ]
    }
    role="menu"
    style={Object {}}
  >
    <li
      aria-disabled={undefined}
      className="ant-menu-item"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      role="menuitem"
      style={Object {}}
      title={undefined}
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "<b>First Item</b>",
          }
        }
      />
    </li>
    <li
      aria-disabled={undefined}
      className="ant-menu-item"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      role="menuitem"
      style={Object {}}
      title={undefined}
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Second Item",
          }
        }
      />
    </li>
    <li
      aria-disabled={undefined}
      className="ant-menu-item"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      role="menuitem"
      style={Object {}}
      title={undefined}
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "Third Item",
          }
        }
      />
    </li>
  </ul>
</div>
`;
